libs/payments/eligibility/src/lib/eligibility.service.ts

```diff
@@ -64,4 +66,56 @@ export class EligibilityService {
 
     return eligibility;
   }
+
+  async getProductAvailabilityForLocation(
+    offeringId: string,
+    countryCode?: string
+  ) {
+    if (!countryCode) {
+      return {
+        status: LocationStatus.Unresolved,
+        message: 'Country code was not resolved',
+      };
+    }
+
+    const { countries } =
+      await this.productConfigurationManager.fetchCMSData(offeringId);
+    const supportedCountries = countries.map((country) => country.slice(0, 2));
+
+    const isSanctionedLocation = countryCode
+      ? this.locationConfig.subscriptionsUnsupportedLocations.includes(
+          countryCode
+        )
+      : undefined;
+
+    const isSupportedLocation = countryCode
+      ? supportedCountries.includes(countryCode)
+      : undefined;
+
+    if (!isSanctionedLocation && isSupportedLocation) {
+      return {
+        status: LocationStatus.Valid,
+        message: `Customer is in a location where the product is supported: ${countryCode}`,
+      };
+    }
+
+    if (isSanctionedLocation) {
+      return {
+        status: LocationStatus.SanctionedLocation,
+        message: `Customer is in a sanctioned location: ${countryCode}`,
+      };
+    }
+
+    if (isSupportedLocation === false) {
+      return {
+        status: LocationStatus.ProductNotAvailable,
+        message: `Product is not available in customer's location: ${countryCode}`,
+      };
+    }
+
+    return {
+      status: LocationStatus.Unresolved,
```

Comment:
Should `Unresolved` be the default state? If it gets to this point, doesn't it mean that countryCode is valid?